akka-stream/src/main/scala/akka/stream/impl/fusing/Ops.scala

```diff
@@ -394,6 +396,62 @@ final case class Fold[In, Out](zero: Out, f: (Out, In) ⇒ Out) extends GraphSta
 /**
  * INTERNAL API
  */
+final class FoldAsync[In, Out](zero: ⇒ Out, f: (Out, In) ⇒ Future[Out]) extends GraphStage[FlowShape[In, Out]] {
```

Comment:
It leads one extra allocation (for the by-name function) that will always be evaluated if the graph stage is materialized, so it doesn't really make sense here.